src/main/java/com/cloudbees/jenkins/support/SupportAction.java

```diff
@@ -185,6 +181,81 @@ public void doDeleteBundles(StaplerRequest req, StaplerResponse rsp) throws Serv
+            fileToDownload = createZipFile(bundlesToDownload);
+        } else {
+            fileToDownload = new File(SupportPlugin.getRootDirectory(), bundlesToDownload.iterator().next());
+        }
+        logger.fine("Trying to download file "+ fileToDownload.getAbsolutePath());
+        try {
+            rsp.setContentType("application/zip");
+            rsp.addHeader("Content-Disposition", "inline; filename=" + fileToDownload.getName() + ";");
+            FileUtils.copyFile(fileToDownload, rsp.getOutputStream());
+            logger.info("Bundle " + fileToDownload.getAbsolutePath() + " successfully downloaded");
+        } catch (RuntimeException e) {
+            logger.log(Level.SEVERE, "Unable to download file " + fileToDownload.getAbsolutePath(), e);
+        } finally {
+            if (bundlesToDownload.size() > 1) {
+                if (fileToDownload.delete()) {
+                    logger.log(Level.FINE, "Temporary multiBundle file deleted: " + fileToDownload.getAbsolutePath());
+                } else {
+                    logger.log(Level.SEVERE, "Unable to delete temporary multiBundle file: " + fileToDownload.getAbsolutePath());
+                }
+            }
+        }
+    }
+
+    private Set<String> getSelectedBundles(StaplerRequest req, JSONObject json) throws ServletException, IOException {
+        Set<String> bundles = new HashSet<>();
+        List<String> existingBundles = getBundles();
+        for (Selection s : req.bindJSONToList(Selection.class, json.get("bundles"))) {
+            if (s.isSelected()) {
+                if (existingBundles.contains(s.getName())) {
+                    bundles.add(s.getName());
+                } else {
+                    logger.log(Level.FINE, "The bundle selected {0} does not exist, so it will not be processed", s.getName());
+                }
+            }
+        }
+        return bundles;
+    }
+
+    private File createZipFile(Set<String> bundles) throws IOException {
+        File rootDirectory = SupportPlugin.getRootDirectory();
+        File zipFile = File.createTempFile(
+            String.format("multiBundle(%s)-", bundles.size()), ".zip");
+
+        try(FileOutputStream fos = new FileOutputStream(zipFile);
+            ZipOutputStream zos = new ZipOutputStream(fos)) {
+            byte[] buffer = new byte[1024]; 
+            for (String bundle: bundles) {
+                File file = new File(rootDirectory, bundle);
+                try(FileInputStream fis = new FileInputStream(file)) {
+                    zos.putNextEntry(new ZipEntry(file.getName()));
+                    int length;
+                    while ((length = fis.read(buffer)) > 0) {
+                        zos.write(buffer, 0, length);
+                    }
+                }
+            }
+        } catch (IOException e) {
+            logger.log(Level.SEVERE, "Error creating zip file: " + zipFile.getAbsolutePath(), e);
+        }
+        zipFile.deleteOnExit();
```

Comment:
@dariver you should do this right after creating the file, before the `try {} catch {}`